addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -4303,6 +4305,60 @@ QUnit.module("Views", (hooks) => {
         assert.verifySteps(["get_views", "read", "write", "read", "execute_action"]);
     });
 
+    QUnit.test('buttons with attr "special" in dialog close the dialog', async function (assert) {
+        serverData.views = {
+            "product,false,form": `
+                <form>
+                    <sheet>
+                        <field name="name" />
+                    </sheet>
+                    <footer>
+                        <button class="btn btn-primary" special="save" data-hotkey="s">Special button save</button>
+                        <button class="btn btn-secondary" special="cancel" data-hotkey="j">Special button cancel</button>
+                    </footer>
+                </form>
+            `,
+        };
+
+        await makeView({
+            type: "form",
+            serverData,
+            resModel: "partner",
+            resId: 2,
+            arch: `
+                <form>
+                    <sheet><group>
+                    <field name="product_id"/>
+                    </group></sheet>
+                </form>`,
+            mockRPC(route, args) {
+                if (args.method === "get_formview_id") {
```

Comment:
I'd assert step here, to record the write rpcs we do

Reply:
I improved the test and asserted the "create" and "write" calls.

- First write ABCDE and save it -> create
- Save the form view -> write
- Write XYZ but cancel creation in the dialog -> not created
- form view has nothing to save